### PR TITLE
android-studio: remove gnome_vfs & GConf

### PR DESCRIPTION
**Pull request by jtojnar**
## Description of changes

Those libraries have been deprecated for 10+ years and require daemons that no-one has running these days.

Continuation of #322599
Part of #39976

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/android-studio/common.nix [hunk 1]
```diff
@@ -175,9 +175,7 @@ let
 
           # For GTKLookAndFeel
           gtk2
```
Comment (jtojnar) [inline pkgs/applications/editors/android-studio/common.nix:177]: We should really switch to GTK 3 but I will leave it to someone else.

Comment (SuperSandro2000) [reply pkgs/applications/editors/android-studio/common.nix:177]: I have gtk2 marked broken in my local config already 😅

**Review by SuperSandro2000 — APPROVED** [review]

**Review by alapshin — APPROVED** [review]

**Review by JohnRTitor — APPROVED** [review]
Approved, @jtojnar do you want to change it to `gtk3` here or should I just merge it?

Ideally I would like this closer to [jetbrains](https://github.com/NixOS/nixpkgs/blob/2893f56de08021cffd9b6b6dfc70fd9ccd51eb60/pkgs/applications/editors/jetbrains/bin/linux.nix#L91), android-studio is a defacto jetbrains product anyway.

**Comment by jtojnar** [comment]
GTK 3 will probably require more testing so best leaving it for a separate PR.

**Pull request merged**